src/System.Text.Json/System/Text/Json/JsonParser.cs

```diff
@@ -190,7 +190,7 @@ public JsonObject Parse(ReadOnlySpan<byte> utf8Json, BufferPool pool = null)
                         break;
                     case JsonTokenType.ArrayEnd:
                         _db.Span.Slice(FindLocation(_stack.ArrayStackCount - 1, false)).Write<int>(arrayItemsCount);
-                        arrayItemsCount += _stack.PopArray();
+                        arrayItemsCount = _stack.PopArray();
```

Comment:
What motivated this change? If this is a bug, can you please add a test that exposes it?